crates/bevy_solari/src/realtime/specular_gi.wgsl

```diff
@@ -78,26 +84,40 @@ fn trace_glossy_path(initial_ray_origin: vec3<f32>, initial_wi: vec3<f32>, rng:
         if ray.kind == RAY_QUERY_INTERSECTION_NONE { break; }
         let ray_hit = resolve_ray_hit_full(ray);
 
-        // Add world cache contribution
-        let diffuse_brdf = ray_hit.material.base_color / PI;
-        radiance += throughput * diffuse_brdf * query_world_cache(ray_hit.world_position, ray_hit.geometric_world_normal, view.world_position, WORLD_CACHE_CELL_LIFETIME, rng);
-
-        // Surface is very rough, terminate path in the world cache
-        if ray_hit.material.roughness > 0.1 && i != 0u { break; }
-
-        // Sample new ray direction from the GGX BRDF for next bounce
         let TBN = calculate_tbn_mikktspace(ray_hit.world_normal, ray_hit.world_tangent);
         let T = TBN[0];
         let B = TBN[1];
         let N = TBN[2];
+
         let wo = -wi;
         let wo_tangent = vec3(dot(wo, T), dot(wo, B), dot(wo, N));
+
+        // Add emissive contribution
+        radiance += throughput * emissive_mis_weight(pdf, ray_hit, surface_perfectly_specular) * ray_hit.material.emissive;
+
+        // Should not perform NEE for mirror-like surfaces
+        surface_perfectly_specular = ray_hit.material.roughness <= 0.001 && ray_hit.material.metallic > 0.9999;
```

Comment:
This should probably go down where everything else (`wi`, `ray_origin`, `pdf` etc) is updated.